encoder/vaapiencoder_base.cpp

```diff
@@ -285,10 +286,9 @@ SurfacePtr VaapiEncoderBase::createNewSurface(uint32_t fourcc)
     switch(fourcc) {
     case VA_FOURCC_NV12:
     case VA_FOURCC_I420:
-        rtFormat = VA_RT_FORMAT_YUV420;
-        break;
+    case VA_FOURCC_P010:
     case VA_FOURCC_YUY2:
-        rtFormat = VA_RT_FORMAT_YUV422;
+        rtFormat = getRtFormat(fourcc);
```

Comment:
please remove entire switch case and use getRtFormat instead and check the return value